include/aikido/control/InstantaneousTrajectoryExecutor.hpp

```diff
@@ -0,0 +1,53 @@
+#ifndef AIKIDO_CONTROL_INSTANTANEOUSTRAJECTORYEXECUTOR_HPP_
+#define AIKIDO_CONTROL_INSTANTANEOUSTRAJECTORYEXECUTOR_HPP_
+
+#include <future>
+#include <mutex>
+
+#include <dart/dynamics/Skeleton.hpp>
+#include <aikido/control/TrajectoryExecutor.hpp>
+#include <aikido/trajectory/Trajectory.hpp>
+
+namespace aikido {
+namespace control {
+
+/// Instantaneously executes a trajectory in simulation by setting DOF positions
+/// to the end of the trajectory.
+class InstantaneousTrajectoryExecutor : public TrajectoryExecutor
+{
+public:
+  /// Constructor.
+  /// \param skeleton Skeleton to execute trajectories on.
+  ///        All trajectories must have dofs only in this skeleton.
+  InstantaneousTrajectoryExecutor(::dart::dynamics::SkeletonPtr skeleton);
```

Comment:
DART `MetaSkeleton`s don't come with an associated mutex, so we wouldn't be able to lock that mutex while checking whether the DOFs specified by the trajectory are in the skeleton. It seems necessary to lock the mutex while doing that. I'd otherwise be happy to use `MetaSkeleton` over `Skeleton`.

@jslee02 why don't `MetaSkeleton`s have mutexes?

Reply:
I don't think we have a good reason not to have a mutex in `MetaSkeleton`. Probably the mutex in `Skeleton` was added when `MetaSkeleton` wasn't yet introduced, and we just didn't add one to `MetaSkeleton` in introducing it?

@mxgrey Do you know why?

If we want to add a mutex to `MetaSkeleton`, I think we can just move the mutex in `Skeleton` to `MetaSkeleton`.